lms/djangoapps/courseware/tests/test_credit_requirements.py

```diff
@@ -133,6 +133,26 @@ def test_credit_requirements_not_eligible(self):
         )
         self.assertContains(response, "Verification Failed")
 
+    def test_credit_requirements_honor_enrollment(self):
+        # Test the progress table is not displayed to the honor students.
+        credit_api.set_credit_requirement_status(
+            self.user.username, self.course.id,
+            "reverification", "midterm",
+            status="failed", reason={}
+        )
+        self.enrollment.mode = 'honor'
+        self.enrollment.save()  # pylint: disable=no-member
+
+        # Check the progress page display
+        response = self._get_progress_page()
+        self.assertNotContains(response, self.MIN_GRADE_REQ_DISPLAY)
+        self.assertNotContains(response, self.VERIFICATION_REQ_DISPLAY)
+        self.assertNotContains(
+            response,
+            "{}, you are no longer eligible for credit in this course.".format(self.USER_FULL_NAME)
+        )
+        self.assertNotContains(response, "Verification Failed")
```

Comment:
These assertions are all negative, but very specific, which means a handful of minor modifications to the template could cause the test to always pass (even if the mode-switch logic was broken).

You could make the test stronger, and also simplify the code, by just checking whether the "credit-eligibility" section element is present in the response content or not.  (cf. https://github.com/edx/edx-platform/blob/aamir-khan/ECOM-3107-credit-eligibility-on-progress/lms/templates/courseware/progress.html#L103)